frontend/javascript/README.md

```diff
@@ -0,0 +1,713 @@
+```js
+// true!!!
+true
+// false...
+false
+// logical or, yields true
+true || false
+// logical and, yields false
+true && false
+// logical not, yields false
+!true
+// use parenthesis to enforce an order of operations
+// this yields true
+(true || false) || (true && false)
+```
+
+Numbers are stored as double-precision floating point numbers (think `double` in your C-style language of choice). They look like:
+```js
+// plain integers yield their value
+1
+// you can add a point without anything after it
+2.
+// or with something
+3.0
+// scientific notation works - this translates to 400 * 10^(-2)
+400e-2
+// numbers prefixed with 0x are interpreted as base-16 (hexadecimal)
+0x5
+// a 0b prefix denotes a base-2 literal (binary)
+0b110
+// you can separate digits with an underscore (_) for readability
+// this yields 1000000 (one million)
+1_000_000
+// addition, yields 3
+1 + 2
+// subtraction, yields -1
+1 - 2
+// multiplication, yields 2
+1 * 2
+// division, yields 0.5
+1 / 2
+// modulo, yields 1
+1 % 2
+// exponentiation (think a^b), yields 1
+1 ** 2
+```
+Please keep in the back of your mind that these are floats, and come with all the usual float weirdness (imprecision, non-associativity, signed zeroes, infinities, NaNs). You should almost never have to worry about these things, but it's good to know that they exist just in case you ever do.
+
+Strings look like this:
+```js
+// double quotes
+"this is a string"
+// single quotes
+'this is also a string'
+// backticks (that thing to the left of your 1 key)
+`this is also also a string`
+// you can use a backslash (\) to break long strings into multiple lines
+// this yields "this string spans multiple lines!" (no new line)
+"this string spans \
+multiple lines!"
+// you can use the usual escape sequences
+"newline: \n, tab: \t, carriage return: \r, null: \0, etc"
+// strings are arrays of individual characters. you can access these characters
+// with a subscript ([]). we'll talk more about arrays later!
+// this yields the string '3'
+"012345"[3]
+// + appends strings together. this yields "strings are nice".
+"strings " + "are nice"
+// other types are eagerly converted into strings when appended
+// this yields "this should say false: false"
+"this should say false: " + (false || (false && true))
+// backtick strings allow you to inject any js expression into a string with
+// `${expr}`
+`this should say false: ${false || (false && true)}`
+```
+
+Undefined and null values are kind of special. Undefined is used to indicate that an identifier hasn't been given an explicit value. You'll see undefined values if you try to do something like use a variable that hasn't been assigned to. Null values, on the other hand, are used to indicate an intentional value of 'nothing'. Most of the time, you shouldn't go out of your way to use undefined. Null is maybe a little bit more common.
+```js
+undefined
+null
+```
```

Comment:
Do we want to cover operations on `null` and `undefined` values?

Reply:
Unless they produce a behavior different than what someone with basic CS knowledge might expect (e.g. a value instead of an error), I think this omission is alright.

Reply:
This is JavaScript, of course they do something unexpected. Jokes aside, I'll include a general statement on how operations can produce weird results and some very brief examples. I don't think it's important enough to go into much more detail than that.